core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2193,17 +2211,8 @@ private void checkStorageTableNotRedirected(QualifiedObjectName source)
             });
         }
 
-        private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, Optional<TableHandle> tableHandle, List<Field> fields, String authorization)
+        private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, RelationType relationType, Scope accessControlScope)
```

Comment:
Why did this change? It seems unrelated to this new feature.

Reply:
The same `Scope` instance needs to be used between row filters and check constraints for the subsequent `Analysis#registerTable`. Otherwise, `PlanSanityChecker` throws "Unexpected identifier in logical plan". Do you have alternative suggestions to avoid this change?